### PR TITLE
create cars model

### LINKED ISSUE #7: [model] cars table
this model will have the following:
  - brand type string
  - customer_id type number
  - model type string
  - price type double
  - year type number 
  - mileage type number 
  - quality type number 
  - is_good_price type boolean 
  - location type string 
  - state type string 
  - transmission type string 
  - features type JSON 
  - description type string 
  - fuel type string 
cars table has a relation with customers table many to one and has a relation   one to many with car_images table

### PR DESCRIPTION
**Pull request by husamkamal**

### CODE REVIEW

**Review by abdou059 — APPROVED** [review]

**Review by mai-mohammed — APPROVED** [review]

File: src/models/Cars.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    primaryKey: true
+  },
+  brand: {
+    type: DataTypes.STRING
+  },
+  model: {
+    type: DataTypes.STRING
+  },
+  price: {
+    type: DataTypes.DOUBLE
```
Comment (LinaYahya) [inline src/models/Cars.ts]:
```suggestion
    type: DataTypes.DOUBLE,
    allowNull: false,
    ```

File: src/models/Cars.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import sequelize from '../config/connection';
+
+const Cars = sequelize.define('cars', {
+  id: {
+    type: DataTypes.INTEGER,
+    autoIncrement: true,
+    primaryKey: true
+  },
+  brand: {
+    type: DataTypes.STRING
```
Comment (LinaYahya) [inline src/models/Cars.ts]:
```suggestion
        allowNull: false,
```

File: src/models/Cars.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  model: {
+    type: DataTypes.STRING
```
Comment (LinaYahya) [inline src/models/Cars.ts]:
```suggestion
    type: DataTypes.STRING,
        allowNull: false,
```

File: src/models/Cars.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    type: DataTypes.INTEGER
+  },
+  isGoodPrice: {
+    type: DataTypes.BOOLEAN
+  },
+  location: {
+    type: DataTypes.STRING
+  },
+  state: {
+    type: DataTypes.STRING
```
Comment (LinaYahya) [inline src/models/Cars.ts]:
```suggestion
        allowNull: false,
         type: DataTypes.ENUM('pending', 'under-check', 'on-market', 'sold'), // options we have, which I don't remember
        defaultValue: 'pending' 

```

File: src/models/Cars.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  mileage: {
+    type: DataTypes.INTEGER
+  },
+  quality: {
+    type: DataTypes.INTEGER
+  },
+  isGoodPrice: {
+    type: DataTypes.BOOLEAN
+  },
+  location: {
```
Comment (LinaYahya) [inline src/models/Cars.ts:36]: allowNull as false

File: src/models/Cars.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    type: DataTypes.STRING
+  },
+  state: {
+    type: DataTypes.STRING
+  },
+  transmission: {
+    type: DataTypes.STRING
+  },
+  features: {
+    type: DataTypes.STRING
```
Comment (LinaYahya) [inline src/models/Cars.ts]:
```suggestion
type: DataTypes.ARRAY(DataTypes.STRING) // not sure if it's string or numbers
```

File: src/models/Cars.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    type: DataTypes.BOOLEAN
+  },
+  location: {
+    type: DataTypes.STRING
+  },
+  state: {
+    type: DataTypes.STRING
+  },
+  transmission: {
+    type: DataTypes.STRING
```
Comment (LinaYahya) [inline src/models/Cars.ts]:
```suggestion
    type: DataTypes.ENUM('auto', 'manual'),
```

**Review by tariqabusada — APPROVED** [review]

File: src/models/Cars.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  features: {
+    type: DataTypes.STRING
+  },
+  description: {
+    type: DataTypes.STRING
+  },
+  fuel: {
+    type: DataTypes.STRING
```
Comment (LinaYahya) [inline src/models/Cars.ts]:
```suggestion
    type: DataTypes.ENUM('', ''), // add options here
```

File: src/models/Cars.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  brand: {
+    type: DataTypes.STRING
+  },
+  model: {
+    type: DataTypes.STRING
+  },
+  price: {
+    type: DataTypes.DOUBLE
+  },
+  year: {
```
Comment (LinaYahya) [inline src/models/Cars.ts:22]: not sure it will be required or not

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**